### ISSUE TITLE
The Main Thread rendering engine doesn't currently support repeaters. To play an animation with repeaters, you can use the Core Animation rendering engine instead

### ISSUE THREAD

**Issue by kanji-dubizzle** [airbnb/lottie-ios#1739]
Hi, I am getting this error.

> Lottie/ItemsExtension.swift:83: Fatal error: The Main Thread rendering engine doesn't currently support repeaters.
> To play an animation with repeaters, you can use the Core Animation rendering engine instead.
> 2022-09-05 21:45:24.501920+0400 Demo[46949:6094753] Lottie/ItemsExtension.swift:83: Fatal error: The Main Thread rendering engine doesn't currently support repeaters.
> To play an animation with repeaters, you can use the Core Animation rendering engine instead.

## Which Version of Lottie are you using?
Lottie 3.4.3

## Animation JSON
https://lottiefiles.com/59461-heart-beat-pop-up

<img width="1000" alt="Screen Shot 2022-09-05 at 9 48 53 PM" src="https://user-images.githubusercontent.com/81670605/188497171-40b8fd39-a83f-4ede-bc74-451306599d9e.png">

**Comment by FreestyleRR** [comment]
I have the same issue

**Comment by FreestyleRR** [comment]
> Hi, I am getting this error.
> 
> > Lottie/ItemsExtension.swift:83: Fatal error: The Main Thread rendering engine doesn't currently support repeaters.
> > To play an animation with repeaters, you can use the Core Animation rendering engine instead.
> > 2022-09-05 21:45:24.501920+0400 Demo[46949:6094753] Lottie/ItemsExtension.swift:83: Fatal error: The Main Thread rendering engine doesn't currently support repeaters.
> > To play an animation with repeaters, you can use the Core Animation rendering engine instead.
> 
> ## Which Version of Lottie are you using?
> Lottie 3.4.3
> 
> ## Animation JSON
> https://lottiefiles.com/59461-heart-beat-pop-up
> 
> <img alt="Screen Shot 2022-09-05 at 9 48 53 PM" width="1000" src="https://user-images.githubusercontent.com/81670605/188497171-40b8fd39-a83f-4ede-bc74-451306599d9e.png">

I fixed this issue.
Try this:
AppDelegate -> LottieConfiguration.shared.renderingEngine = .automatic

**Comment by kanji-dubizzle** [comment]
@FreestyleRR which Animation are you using?

Still, it is not working for me. For the mentioned above json.

**Comment by FreestyleRR** [comment]
> @FreestyleRR which Animation are you using?
> 
> Still, it is not working for me. For the mentioned above json.

Json animation and I have the same issue
But it helped me.
Try this -
 let animationView = AnimationView(
  animation: ...,
  configuration: LottieConfiguration(renderingEngine: .automatic)
)

**Comment by kanji-dubizzle** [comment]
```
    let animation = Animation.named("favourite_car", bundle: .main, subdirectory: nil, animationCache: nil)
    self.animationView = AnimationView(animation: animation,
                                       configuration: LottieConfiguration(renderingEngine: .automatica))
    animationView!.frame = animatedView.bounds
    animationView!.contentMode = .scaleAspectFit
    animationView!.loopMode = .loop
    animationView!.animationSpeed = 0.5
    animatedView.addSubview(animationView!)
    animationView!.play()
```

@FreestyleRR above is the code snippet, but still same crash.

**Comment by Ronaldoh1** [comment]
I've got the same issue. 

Lottie/ItemsExtension.swift:83: Fatal error: The Main Thread rendering engine doesn't currently support repeaters.
To play an animation with repeaters, you can use the Core Animation rendering engine instead.

just trying to render on a viewController...all it's doing is presenting the animation view.

**Comment by AlexanderKvamme** [comment]
I am also experiencing this crash while instantiating the animationView like this:
`AnimationView(animation: Lottie.lottieLoadingIndicator, configuration: LottieConfiguration(renderingEngine: .automatic)).
`
Is there anything else that needs to be done?

**Comment by calda** [comment]
We could change this to be a warning that prints to console (`LottieLogger.shared.warn`) instead of an assertion that crashes in debug (`LottieLogger.shared.assertionFailure`). It seems like there were a lot of folks using animations with silently-hidden repeaters, and we shouldn't break their use case. 

Is anyone interested in making a pull request for this change? It should be a one-liner 😄

**Comment by kanji-dubizzle** [comment]
![Screen Recording 2022-09-20 at 12 07 11 PM](https://user-images.githubusercontent.com/81670605/191202454-1230c0fb-ac81-4944-9151-5a75bb6b4365.gif)

Even after this the animation is not loaded correctly.

**Comment by calda** [comment]
Yes, https://github.com/airbnb/lottie-ios/pull/1754 only prevents the "unsupported" warning message from crashing the app in debug builds. 

@kanji-dubizzle, have you tried playing your animation with the [Core Animation rendering engine](https://github.com/airbnb/lottie-ios/discussions/1627)? That implementation supports repeaters.

**Comment by kanji-dubizzle** [comment]
@calda 
![Screen Shot 2022-09-23 at 2 21 34 PM](https://user-images.githubusercontent.com/81670605/191970276-1481d62c-fe33-41fc-8a1b-1674d23ab90f.png)

**Comment by calda** [comment]
It looks like your animation uses features that aren't supported yet by the Core Animation rendering engine. What version of Lottie are you using? Could you share a copy of your animation?

**Comment by kanji-dubizzle** [comment]
3.4.2.
Link to JSON: **https://lottiefiles.com/59461-heart-beat-pop-up**

**Comment by calda** [comment]
Could you test your animation on the latest version, `3.4.3`?

**Comment by kanji-dubizzle** [comment]
Still it's same
<img width="584" alt="Screen Shot 2022-09-26 at 11 14 51 AM" src="https://user-images.githubusercontent.com/81670605/192206057-aa16cd23-47c1-4a6d-8f07-ccd8d70aec6c.png">

**Comment by jorgealegre** [comment]
I'm also having the same problem with an animation file. What makes an animation not work with Core Animation?

**Comment by kirtiios** [comment]
i have same problem ,any one find solution

**Comment by jorgealegre** [comment]
We ended up replacing our animation with one that looks exactly the same but with a different implementation from the designer and got really good results.

**Comment by bogdan-razvan** [comment]
I was having the same issue while using Lottie 3.4.3.

Updating to 4.2.0 solved the issue.
